src/main/java/org/apache/pirk/querier/wideskies/encrypt/EncryptQuery.java

```diff
@@ -56,103 +56,84 @@
-  // if the selector is of variable lengths
-  private HashMap<Integer,String> embedSelectorMap = null;
-
-  public EncryptQuery(QueryInfo queryInfoInput, List<String> selectorsInput, Paillier paillierInput)
-  {
-    queryInfo = queryInfoInput;
-
-    selectors = selectorsInput;
-
-    paillier = paillierInput;
-
-    embedSelectorMap = new HashMap<>();
-  }
-
-  public Paillier getPaillier()
-  {
-    return paillier;
-  }
-
-  public QueryInfo getQueryInfo()
-  {
-    return queryInfo;
-  }
-
-  public Query getQuery()
-  {
-    return query;
-  }
-
-  public Querier getQuerier()
-  {
-    return querier;
-  }
-
-  public List<String> getSelectors()
-  {
-    return selectors;
-  }
-
-  public HashMap<Integer,String> getEmbedSelectorMap()
+  /**
+   * Constructs a query encryptor using the given query information, selectors, and Paillier cryptosystem.
+   * 
+   * @param queryInfo
+   *          Fundamental information about the query.
+   * @param selectors
+   *          the list of selectors for this query.
+   * @param paillier
+   *          the Paillier cryptosystem to use.
+   */
+  public EncryptQuery(QueryInfo queryInfo, List<String> selectors, Paillier paillier)
   {
-    return embedSelectorMap;
+    this.queryInfo = queryInfo;
+    this.selectors = selectors;
+    this.paillier = paillier;
   }
 
   /**
-   * Encrypt, building the Query object, calculating and setting the query vectors.
+   * Encrypts the query described by the query information using Paillier encryption.
+   * <p>
+   * The encryption builds a <code>Querier</code> object, calculating and setting the query vectors.
    * <p>
    * Uses the system configured number of threads to conduct the encryption, or a single thread if the configuration has not been set.
    * 
    * @throws InterruptedException
-   *           if the task was interrupted during encryption.
+   *           If the task was interrupted during encryption.
    * @throws PIRException
+   *           If a problem occurs performing the encryption.
+   * @return The querier containing the query, and all information required to perform decryption.
    */
-  public void encrypt() throws InterruptedException, PIRException
+  public Querier encrypt() throws InterruptedException, PIRException
   {
     int numThreads = SystemConfiguration.getIntProperty("numThreads", 1);
-    encrypt(numThreads);
+    return encrypt(numThreads);
   }
 
   /**
-   * Encrypt, building the Query object, calculating and setting the query vectors
-   * <p>
-   * If we have hash collisions over our selector set, we will append integers to the key starting with 0 until we no longer have collisions.
-   * <p>
-   * For encrypted query vector E = <E_0, ..., E_{(2^hashBitSize)-1}>:
+   * Encrypts the query described by the query information using Paillier encryption using the given number of threads.
    * <p>
-   * E_i = 2^{j*dataPartitionBitSize} if i = H_k(selector_j) 0 otherwise
```

Comment:
Sure!  Any thoughts on how we can get this to render a bit better for the reader?

I removed it because it seems like an implementation detail rather than an API usage / spec description; and the markup [is not rendered well in javadoc](http://pirk.incubator.apache.org/javadocs/org/apache/pirk/querier/wideskies/encrypt/EncryptQuery.html#encrypt(int)).

Putting it back now.